index.js

```diff
@@ -3,20 +3,48 @@ var minimatch = require('minimatch');
 
 module.exports = main;
 
-function main(plugins) {
+function main(plugins, options) {
   var plugins = plugins || [];
+  var options = options || {};
+  var map = normalizeMapOptions(options.map);
 
-  var processor = postcss();
-
-  plugins.forEach(function(plugin) {
-    processor.use(plugin);
-  });
+  var processor = postcss(plugins);
 
   return function (files, metalsmith, done) {
     var styles = Object.keys(files).filter(minimatch.filter("*.css", { matchBase: true }));
-    styles.forEach(function (file, index, arr) {
-      files[file].contents = new Buffer(processor.process(files[file].contents.toString()).css);
+
+    styles.forEach(function (file) {
```

Comment:
You are right, it's very probable that processing is continued while the plugin already signalled that it's done.